### ISSUE TITLE
Allow differentiating between number and integer

### ISSUE THREAD

**Issue by joca-bt** [lukas-krecan/JsonUnit#218]
Although integer is not part of the JSON Schema data model (from the spec: _Note that JSON Schema vocabularies are free to define their own extended type system. This should not be confused with the core data model types defined here. As an example, "integer" is a reasonable type for a vocabulary to define as a value for a keyword, but the data model makes no distinction between integers and other numbers._) in practice it is heavily used in vocabulary and allowed as a type by several applications, libraries, and frameworks (e.g. http://fasterxml.github.io/jackson-core/javadoc/2.10/com/fasterxml/jackson/core/JsonToken.html). For this reason, it would be helpful to be able distinguish between them, i.e. having an `isNumber()` and an `isInteger()` in `JsonAssert`.

**Comment by lukas-krecan** [comment]
Thanks for feedback. It's a bit tricky, since JsonUnit is able to parse JSON using different libraries and not all expose such information. Will take a look at it.

**Comment by lukas-krecan** [comment]
Would this work for you https://github.com/lukas-krecan/JsonUnit/pull/220/files?

**Comment by joca-bt** [comment]
It would!

**Comment by lukas-krecan** [comment]
Just to make sure that I understand you correctly, would you consider "1.0" to be an integer or not? In other words, do you want to check absence of the decimal point or that the number is integer?

**Comment by joca-bt** [comment]
1.0 in the json body would be a number and not an integer.

**Comment by lukas-krecan** [comment]
Ok, and what about "1e3"?

**Comment by lukas-krecan** [comment]
I think this should do 

```java
    /**
     * Asserts that the value is an integer. 1 is an integer 1.0, 1.1, 1e3, 1e0, 1e-3 is not.
     */
    public BigIntegerAssert isIntegralNumber()
```

**Comment by joca-bt** [comment]
That would be great!

**Comment by lukas-krecan** [comment]
Released as 2.14.0
